src/dagre/rank/network-simplex.js

```diff
@@ -173,20 +173,22 @@ function enterEdge(t, g, edge) {
     flip = true;
   }
 
-  var candidates = g.edges().filter((edge) => {
+  const candidates = g.edges().filter((edge) => {
     return (
       flip === isDescendant(t, t.node(edge.v), tailLabel) &&
       flip !== isDescendant(t, t.node(edge.w), tailLabel)
     );
   });
 
-  return candidates.reduce((acc, edge) => {
-    if (slack(g, edge) < slack(g, acc)) {
-      return edge;
-    }
+  if (candidates.length > 0) {
+    return candidates.reduce((acc, edge) => {
+      if (slack(g, edge) < slack(g, acc)) {
+        return edge;
+      }
 
-    return acc;
-  });
+      return acc;
+    });
+  }
```

Comment:
In the future, we could also add the following, to make it a bit more explicit that `enterEdge` can return `undefined`, but if ESLint isn't giving us a warning/error, it's fine for now.

```suggestion
  } else {
    return undefined;
  }
```